modules/version-checker/version_checker_test.go

```diff
@@ -0,0 +1,112 @@
+package version_checker
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestParamValidation(t *testing.T) {
```

Comment:
I like how many cases you're testing for here. One NIT to consider: this test might be more easily maintained going forward as a ["table driven test"](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) - such that: 

* Quickly scanning the function visually for which cases are implemented and which are not is easier
* It's easier for other engineers in the future to append their own cases, since the overall test harness you've created is handling the test run plumbing itself. They'd only need to define new structs for inputs and the associated expected outputs.